tests/manage/z_cluster/cluster_expansion/test_node_expansion.py

```diff
@@ -30,5 +33,41 @@ def test_add_node(self):
             logger.info(f'The worker nodes number after {len(helpers.get_worker_nodes())}')
 
         else:
-            pytest.skip("UPI not yet supported")
-        # ToDo run IOs
+            new_nodes = 3
+            initial_nodes = helpers.get_worker_nodes()
+            logger.info(f'The worker nodes number before {initial_nodes}')
+            plt = PlatformNodesFactory()
+            node_util = plt.get_nodes_platform()
+            if config.ENV_DATA.get('rhel_workers'):
+                logging.info('AWS UPI RHEL adding node')
+                node_util.create_and_attach_nodes_to_cluster(
+                    {}, 'RHEL', new_nodes)
+            else:
+                logging.info('AWS UPI RHCOS adding node')
+                node_util.create_and_attach_nodes_to_cluster(
+                    {}, 'RHCOS', new_nodes)
+            for sample in TimeoutSampler(
+                timeout=600, sleep=6, func=helpers.get_worker_nodes
+            ):
+                if len(sample) == len(initial_nodes) + new_nodes:
+                    break
+
+            nodes_after_exp = helpers.get_worker_nodes()
+            logger.info(f'The worker nodes number after {len(nodes_after_exp)}')
+            wait_for_nodes_status(
+                node_names=helpers.get_worker_nodes(),
+                status=constants.NODE_READY
+            )
+
+            new_spun_nodes = list(set(nodes_after_exp) - set(initial_nodes)
+                                  )
+            node_obj = ocp.OCP(kind='node')
+
+            for new_spun_node in new_spun_nodes:
+                node_obj.add_label(
+                    resource_name=new_spun_node,
+                    label=constants.OPERATOR_NODE_LABEL
+                )
+                logging.info(
+                    f"Successfully labeled {new_spun_node} with OCS storage label"
+                )
```

Comment:
please create a function for that, in ocs_ci/ocs/node.py

Reply:
ack